ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/BodyValidation.java

```diff
@@ -54,9 +56,13 @@ private static Bytes indexKey(final int i) {
   public static Hash transactionsRoot(final List<Transaction> transactions) {
     final MerklePatriciaTrie<Bytes, Bytes> trie = trie();
 
-    for (int i = 0; i < transactions.size(); ++i) {
-      trie.put(indexKey(i), RLP.encode(transactions.get(i)::writeTo));
-    }
+    IntStream.range(0, transactions.size())
```

Comment:
a for loop will have better performance in this instance, and is more readable.  JVM JITs have been recognizing this pattern for decades and produce very nice code for it.